tests/Dumplie/Spec/Application/Command/Inventory/CreateProductHandlerSpec.php

```diff
@@ -0,0 +1,93 @@
+<?php
+
+namespace Spec\Dumplie\Application\Command\Inventory;
+
+use Dumplie\Application\Command\Inventory\CreateProduct;
+use Dumplie\Application\Transaction\Factory as TransactionFactory;
+use Dumplie\Application\Transaction\Transaction;
+use Dumplie\Domain\Inventory\Product;
+use Dumplie\Domain\Inventory\Products;
+use Dumplie\Domain\SharedKernel\Money\Price;
+use Dumplie\Domain\SharedKernel\Product\SKU;
+use PhpSpec\ObjectBehavior;
+use Prophecy\Argument;
+
+class CreateProductHandlerSpec extends ObjectBehavior
```

Comment:
Personally I don't think that command handlers are good for unit testing (lol I know how it sounds). 
It might not be visible at this particular case but when application is getting bigger handlers are growing (captain obvious FTW!). At some point they are so big you are not able to write any unit tests for them because of large amount of mocks/stubs that needs to be used. Of course you can try to split handler logic between smaller objects but then you end up with a lot of "services" with funny names. 
Thats why I didn't write any specs for handlers that I made and instead of that I created just a single integration [CartTest](https://github.com/dumplie/dumplie/blob/master/tests/Dumplie/Test/Integration/Application/InMemory/Customer/CartTest.php). 
I like to think that command handlers are not the best place for object oriented approach.
Anyway, this is just my experience I'm not trying to force removing those specs we can keep them and decide later if we would like to replace them with integration in-memory tests or not.